### ISSUE TITLE
[scroll-animations-1] Range for entire scroll range

### ISSUE THREAD

**Issue by flackr** [w3c/csswg-drafts#9367]
It has come up a few times (https://github.com/w3c/csswg-drafts/issues/8942#issuecomment-1722261722, https://github.com/w3c/csswg-drafts/issues/8672#issuecomment-1545980510, https://github.com/w3c/csswg-drafts/issues/8578#issuecomment-1540705525) that we don't have any way to specify the entire scroll range for view timelines.

We decided in other issues that because we want the default range for a view timeline to be the cover range we should have cover by the normal range. If we want to preserve this but still support the use cases for the entire scroll range we should add a new range name for view timelines to cover this. E.g. the `scroll` range could represent the full range. Then authors could do something like `animation-range: entry 0% scroll 1000px` to make an animation from when an element enters to when you're scrolled 1000px down.

My preference would be to have `animation-range` without a range name use this range, e.g. just `animation-range: entry 0% 1000px` or `animation-range: 200px 1000px` but I think @fantasai had concerns with having the default animation range be cover except if you specify an animation-range without a named range.

**Comment by ydaniv** [comment]
Big +1 in general.

Regarding:

> My preference would be to have animation-range without a range name use this range, e.g. just animation-range: entry 0% 1000px or animation-range: 200px 1000px but I think @fantasai had concerns with having the default animation range be cover except if you specify an animation-range without a named range.

I'm not sure I'm following.
For now we have no range defaults to the timeline's full range, which is `cover` for View, and `scroll` for Scroll.
IIRC `animation-range: entry 0% 1000px` should expand to `animation-range: entry 0% cover calc(0% + 1000px)`. So if you specify a range without a name it uses default range, right?

**Comment by ydaniv** [comment]
@flackr ok I think I understand your point.
I guess for ViewTimeline having the default be `cover` is more sensible than `scroll`, then `0% 90%` for view would default to an actual view timeline. Otherwise it would default to yet another ScrollTimeline.
So `cover` seems like a good default, and having `scroll` as an opt-in for breaking out of that range is also very useful.

I think this issue is ready for agenda+, WDYT?

**Comment by bramus** [comment]
Two thoughts:

1. Instead of `scroll` why not name it `full`? As that’s what it targets, right?
2. When using no range-name, I think it would be most consistent if it uses the range-name that is used for `normal`.
    - For ViewTimeline that would be `cover`
    - For ScrollTimeline that would be `full`

Some examples:

- VT: `animation-range: 10% 90%` = `animation-range: cover 10% cover 90%`;
- VT: `animation-range: entry 10% 90%` = `animation-range: entry 10% cover 90%`;
- ST: `animation-range: 10% 90%` = `animation-range: full 10% full 90%`;

**Comment by schenney-chromium** [comment]
I think I get why scroll makes sense as a name: You are using the range of a scroll timeline for a view timeline. In which case I am fine with scroll.

**Comment by flackr** [comment]
> bikeshed keyword in issue

To carry out this bikeshedding, the proposals are:
1. 😄 `scroll` as in the *scroll*able range of the associated scrolling container and the range you would get from a *scroll* timeline.
2. 🎉 `full` as in the *full* scrollable range of the associated scrolling container. Confusingly, this can be smaller than the cover range when the scrollable range cannot scroll the subject out of view.
3. ❤️ `content` as in the range of the scrollable *content* in the associated scrolling container, minus the size of the scrollport.

I think we're pretty well aligned on #1, but happy to have emoji votes / comments or rationales for the others, or suggestions for alternatives.

**Comment by bramus** [comment]
If we do name it `scroll`, then that should also be a valid range name for `ScrollTimeline`, right?

That way, using `scroll 0% scroll 100%` as the range on both a `ScrollTimeline` and `ViewTimeline` (that track the same scroller) is the same.

Additionally, for `ScrollTimeline` specifically, all these range values would mean the same:`scroll 0% scroll 100%` = `0% 100%` = `normal normal`.

**Comment by bramus** [comment]
> If we do name it `scroll`, then that should also be a valid range name for `ScrollTimeline`, right?

That would also allow fixing that non spec-compliant situation in Chrome where you have to pass in `"none"` as the name when trying to set a range using an object in JavaScript. If this were `"scroll"` then it’d make sense.

```js
anim.rangeStart: {
  offset: CSS.px(100),
  rangeName: 'none', // This needs to be `"none"`, but `none` is not defined in the spec as a valid `rangeName`.
},
```

**Comment by bramus** [comment]
OK, the poll in https://github.com/w3c/csswg-drafts/issues/9367#issuecomment-1904777189 is clearly leaning towards using `scroll` as the keyword that denominates the full scroll range, so that’s ready to get resolved on. Adding async label for @astearns to pick up.

PROPOSED RESOLUTION: They keyword that represents the entire scroll range is `scroll`.

**Comment by bramus** [comment]
Would love some input on my suggestion in https://github.com/w3c/csswg-drafts/issues/9367#issuecomment-1905788011 to also enable that keyword for `ScrollTimeline`. WDYT @flackr @fantasai?

**Comment by ydaniv** [comment]
Another option to consider could be `fill`, thinking forward if we ever get in the future to something like `pointer-timeline`, as [proposed here](https://github.com/w3c/csswg-drafts/issues/10574) or otherwise, just thinking forward to be consistent.

**Comment by astearns** [comment]
I’m not sure the notion of `fill` as 2d thing translates well to a 1d timeline. Do we need a single word keyword here? Could it be `full-range` or some such?

**Comment by ydaniv** [comment]
Fair enough, maybe it's weird to have `-range` inside there? and we did prefer scroll over full.

**Comment by flackr** [comment]
@bramus having the scroll range work on scroll timeline as well seems reasonable.

**Comment by bramus** [comment]
Closing as I believe the edits are in with the merge of #13306
